axelor-base/src/main/java/com/axelor/apps/base/service/partner/api/GenericApiFetchService.java

```diff
@@ -0,0 +1,76 @@
+package com.axelor.apps.base.service.partner.api;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.PartnerApiConfiguration;
+import com.axelor.apps.base.db.repo.TraceBackRepository;
+import com.axelor.apps.base.exceptions.BaseExceptionMessage;
+import com.axelor.i18n.I18n;
+import com.google.common.net.HttpHeaders;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import wslite.json.JSONException;
+import wslite.json.JSONObject;
+
+public abstract class GenericApiFetchService {
+
+  public abstract String fetch(PartnerApiConfiguration configuration, String identifier)
+      throws AxelorException;
+
+  protected String getUrl(PartnerApiConfiguration configuration, String identifier) {
+    return configuration.getApiUrl() + "/" + identifier;
+  }
+
+  protected String treatResponse(HttpResponse<String> response, String identifier)
```

Comment:
I think this method is useless, as the response will depend on which API is used. So this method should be abstract.